src/module_wrap.cc

```diff
@@ -800,6 +804,65 @@ Maybe<URL> PackageMainResolve(Environment* env,
   return Nothing<URL>();
 }
 
+Maybe<URL> PackageExportsResolve(Environment* env,
+                                 const URL& pjson_url,
+                                 const std::string& pkg_subpath,
+                                 const PackageConfig& pcfg,
+                                 const URL& base) {
+  Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
+  Local<Value> exports = pcfg.exports.Get(isolate);
+  if (exports->IsObject()) {
+    Local<Object> exports_obj = exports.As<Object>();
+    Local<String> subpath = String::NewFromUtf8(isolate,
+        pkg_subpath.c_str(), v8::NewStringType::kNormal).ToLocalChecked();
+
+    auto target = exports_obj->Get(context, subpath).ToLocalChecked();
```

Comment:
Is the `exports` object *guaranteed* to be from JSON at this point (i.e. not a generic user-provided JS object)?

Reply:
I think so, yes. It will always be parsed by the code itself from `package.json`. There shouldn’t be any programmatic way to pass it in.